### ISSUE TITLE
custom frame resize issue, app is working with some rendering issues

### ISSUE THREAD

**Issue by michaelsd28** [JetBrains/compose-jb#1432]
I am trying to make an app with a custom frame and my app is having resizing issues and the cursor is not changing when resizing 

I also get this error on the console **Failed to create DirectX12 device.** i have a RTX 2060 Super

![ezgif com-gif-maker](https://user-images.githubusercontent.com/39009298/142750110-bb1bf668-efa6-49b5-b22d-e0cde9c1a4fd.gif)

this is my code 

![image](https://user-images.githubusercontent.com/39009298/142944025-f79fac13-8d41-4093-b8bd-be9120a31f2b.png)

**Comment by Rsedaikin** [comment]
@michaelsd28 Have you set the transparency property in `Window` to `true`? Are you using 'WindowDraggableArea' as your custom window title bar in this example? Does it have modifier `fillMaxSize()` - expanded to fill entire window?

**Comment by michaelsd28** [comment]
> @michaelsd28 Have you set the transparency property in `Window` to `true`? Are you using 'WindowDraggableArea' as your custom window title bar in this example? Does it have modifier `fillMaxSize()` - expanded to fill entire window?

**Comment by michaelsd28** [comment]
> > @michaelsd28 Have you set the transparency property in `Window` to `true`? Are you using 'WindowDraggableArea' as your custom window title bar in this example? Does it have modifier `fillMaxSize()` - expanded to fill entire window?

this the code i have
`fun main() = application {
    //   window state
    val windowState = rememberWindowState(isMinimized = false)
    Window(
            onCloseRequest = ::exitApplication,
            undecorated = true,
            resizable = true,
            transparent = true,
            state = windowState,
    ) {
        WindowDraggableArea(
                Modifier.fillMaxWidth()
                        .padding(
                                3.dp,
                        ),
        ) {
            Column {
                TopBar().titleBar("Notepad", windowState)
                Body().TextArea()
            }
        }
    }
}
`

**Comment by igordmn** [comment]
The minimal reproducer:
```
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.window.WindowDraggableArea
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication(
    undecorated = true,
) {
    WindowDraggableArea {
        Box(Modifier.fillMaxWidth().height(30.dp).background(Color.Red))
    }
}
```

https://user-images.githubusercontent.com/5963351/143045174-1ed0ae41-919f-43a6-9fe6-f949e8f932c1.mp4


@Rsedaikin , I believe this is because we don't consume events in UndecoratedWindowResizer.

The part of the fix would be:
```
Index: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt b/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt
--- a/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt	(revision fb8d17e307f98832259e59f7a76abd9f616a1666)
+++ b/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt	(date 1637678402119)
@@ -280,18 +280,22 @@
 
     private fun onMouseEvent(event: MouseEvent) {
         lastMouseEvent = event
-        events.post {
-            catchExceptions {
-                scene.onMouseEvent(density, event)
+        if (!event.isConsumed) {
+            events.post {
+                catchExceptions {
+                    scene.onMouseEvent(density, event)
+                }
             }
         }
     }
 
     private fun onMouseWheelEvent(event: MouseWheelEvent) {
         lastMouseEvent = event
-        events.post {
-            catchExceptions {
-                scene.onMouseWheelEvent(density, event)
+        if (!event.isConsumed) {
+            events.post {
+                catchExceptions {
+                    scene.onMouseWheelEvent(density, event)
+                }
             }
         }
     }
Index: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt b/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt
--- a/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt	(revision fb8d17e307f98832259e59f7a76abd9f616a1666)
+++ b/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt	(date 1637678402122)
@@ -71,6 +71,7 @@
         val point = event.getPoint()
         sides = getSides(point)
         fun setCursor(cursorType: Int) {
+            event.consume()
             layer.scene.component.desiredCursor = Cursor(cursorType)
         }
         when (sides) {
```

But we should make sure that these listeners:
```
layer.component.addMouseListener(mouseListener)
layer.component.addMouseMotionListener(motionListener)
```
are triggered before
```
_component.addMouseMotionListener(object : MouseMotionAdapter() {
```

This would also fix this [issue](https://github.com/JetBrains/compose-jb/issues/1444)
```

**Comment by igordmn** [comment]
We should probably also replace this
```
layer.scene.component.desiredCursor
```
by
```
window.setCursor
```
again. because desiredCursor won't be triggered after that.
cc @prepor

**Comment by michaelsd28** [comment]
> We should probably also replace this
> 
> ```
> layer.scene.component.desiredCursor
> ```
> 
> by
> 
> ```
> window.setCursor
> ```
> 
> again. because desiredCursor won't be triggered after that. cc @prepor

The pointer issue was fixed by upgrading to the rc3 version

**Comment by igordmn** [comment]
Fixed in 1.0.0-rc5

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-1432">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
